tensorflow/python/util/nest.py

```diff
@@ -128,7 +136,7 @@ def _sequence_like(instance, args):
   Returns:
     `args` with the type of `instance`.
   """
-  if _is_mapping(instance):
+  if _is_mutable_mapping(instance):
```

Comment:
I'm a bit worried that we might break someone who relied on this behavior before. Let's keep the old branch for _is_mapping (removing the pieces specific to defaultdict), but add a warning to it:

```
tf_logging.log_first_n(
    tf_logging.WARN, "Mapping types may not work well with tf.nest. Prefer using MutableMapping for {}".format(instance_type), 1)
```

Reply:
So shall I create a normal `dict` for for the `is_mapping` branch while adding the warning at the beginning of the branch which would be better?

Reply:
Yes. Preserve the original code, minus the special cases for defaultdict, and adding the warning.